### PR TITLE
ISPN-6119 Query DSL should not accept null or empty parameter name

### PR DESCRIPTION
**Pull request by anistor**
Jira: https://issues.jboss.org/browse/ISPN-6119
        https://issues.jboss.org/browse/ISPN-6121

Also includes all tests from here: https://github.com/infinispan/infinispan/pull/3951

### CODE REVIEW

File: query-dsl/src/main/java/org/infinispan/query/dsl/impl/ParameterExpression.java [hunk 1]
```diff
@@ -11,6 +11,9 @@
    private final String paramName;
 
    public ParameterExpression(String paramName) {
+      if (paramName == null || paramName.isEmpty()) {
+         throw new IllegalArgumentException("Parameter name cannot be null or empty");
```
Comment (tristantarrant) [inline query-dsl/src/main/java/org/infinispan/query/dsl/impl/ParameterExpression.java]: i18n

**Comment by anistor** [comment]
Oh noes! The curse of Babel again! Fixed.

**Comment by anistor** [comment]
DSLQueryIT.testDSLQuery fails (http://ci.infinispan.org/viewLog.html?buildId=34592&tab=buildResultsDiv&buildTypeId=bt9)

This seems to be caused by recent introduction of jboss-logging in query-dsl.

**Comment by anistor** [comment]
Fixed. Let's see what CI says.

**Comment by anistor** [comment]
The CI failures are not related.

**Comment by tristantarrant** [comment]
Pushed, thanks @anistor and @mcimbora

**Pull request closed without merging**